src/Handlers/AbstractPaymentMethodHandler.php

```diff
@@ -331,6 +365,13 @@ public function preparePaymentsRequest(
         //Validate state.data for payment and build request object
         $request = $this->checkoutStateDataValidator->getValidatedAdditionalData($request);
 
+        //Setting payment method type if not present in statedata
+        if (empty($request['paymentMethod']['type'])) {
```

Comment:
when was it empty?

Reply:
When the PM doesn't create a state.data with the generic component, so in that case the request starts as an empty array https://github.com/Adyen/adyen-shopware6/blob/develop/src/Handlers/AbstractPaymentMethodHandler.php#L315